components/p3a/brave_p3a_metric_log_store.h

```diff
@@ -25,25 +25,26 @@ namespace brave {
 // All logs (not only unsent are persistent), and all logs could be loaded
 // using |LoadPersistedUnsentLogs()|. We should fix this at some point since
 // for now persisted entries never expire.
-class BraveP3ALogStore : public metrics::LogStore {
+class BraveP3AMetricLogStore : public metrics::LogStore {
```

Comment:
dont need the inheritance

Reply:
wdyt about discussing this in a separate issue? I would argue that  the interface is well defined and that we benefit from it, as it keeps our log store implementations in line. Also in the message manager `OnLogUploadComplete` we use the interface

Reply:
okay we can postpone it, but I do think we don't need to inherit google-controllable interfaces. We already had to fix our code without good reason, just b/c the upstream changed their code. For sanity, we can extract our own interfaces if we want them